pkg/signing/signing.go

```diff
@@ -360,8 +357,30 @@ func Verify(cpath, url string, id uint32, isGroup bool, authToken string, noProm
 		}
 		authok += fmt.Sprintf("\t%s, KeyID %X\n", name, signer.PrimaryKey.KeyId)
 	}
+	sylog.Infof("Container is signed")
 	fmt.Printf("Data integrity checked, authentic and signed by:\n")
-	fmt.Print(authok)
+	fmt.Printf("%v", authok)
+
+	// if theres no local key, then ask to store it
+	if noLocalKey {
+		if noPrompt {
+			// always store key when prompts disabled
+			if err = sypgp.StorePubKey(netlist[0]); err != nil {
+				return fmt.Errorf("could not store public key: %s", err)
+			}
+		} else {
+			// Ask to store new public key
+			resp, err := sypgp.AskQuestion("Would you like to add this key to your local keyring? [Y/n] ")
+			if err != nil {
+				return err
+			}
+			if resp == "" || resp == "y" || resp == "Y" {
+				if err = sypgp.StorePubKey(netlist[0]); err != nil {
```

Comment:
it looks like this bit of code (storing the public key) is common to both the interactive and the non-interactive paths. Maybe you can write this so that you have something like this:

    if prompt {
        // prompt here
        store = askQuestion(...)
    } else {
        store = true
    }
    if store {
        sypgp.StorePubKey(...)
    }